.github/workflows/reusable-merge.yml

```
@@ -0,0 +1,48 @@
+name: Merge Branch
+
+on:
+  workflow_call:
+    inputs:
+      source_branch:
+        description: 'The source branch to merge from (e.g., main, testnet)'
+        required: true
+        type: string
+      target_branch:
+        description: 'The target branch to merge into (e.g., testnet, mainnet)'
+        required: true
+        type: string
+
+permissions:
+  contents: write
+
+jobs:
+  merge:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Set up Git user
+        run: |
+          git config user.name "github-actions[bot]"
+          git config user.email "github-actions[bot]@users.noreply.github.com"
+
+      - name: Merge and Push
+        run: |
+          echo "Source: ${{ inputs.source_branch }}"
+          echo "Target: ${{ inputs.target_branch }}"
+
+          SOURCE_COMMIT_SHA=$(git rev-parse origin/${{ inputs.source_branch }})
+
+          git checkout ${{ inputs.target_branch }}
+          git pull origin ${{ inputs.target_branch }} --ff-only
+
+          COMMIT_MESSAGE="Merge commit '$SOURCE_COMMIT_SHA' into ${{ inputs.target_branch }}"
+          echo "Using commit message: $COMMIT_MESSAGE"
+
+          git merge --no-ff origin/${{ inputs.source_branch }} -m "$COMMIT_MESSAGE"
+
+          echo "Pushing changes to ${{ inputs.target_branch }}..."
+          git push origin ${{ inputs.target_branch }}
```

Is this adequate? I'm not sure if we can sufficiently protect ourselves against interpolation attacks here (since the `{source,target}_branch`es are just strings and this job has permissions to push, etc.).

```suggestion
          local source="${{ inputs.source_branch }}"
          local target="${{ inputs.target_branch }}"
          
          echo "Source: ${source}"
          echo "Target: ${target}"

          SOURCE_COMMIT_SHA=$(git rev-parse origin/"${source}")

          git checkout "${target}"
          git pull origin "${target}" --ff-only

          COMMIT_MESSAGE="Merge commit '$SOURCE_COMMIT_SHA' into ${target}"
          echo "Using commit message: $COMMIT_MESSAGE"

          git merge --no-ff "origin/${source}" -m "$COMMIT_MESSAGE"

          echo "Pushing changes to ${target}..."
          git push origin "${target}"
```

---

Your are right. I've added branch names as variables and this will prevent interpolation attacks